plugins/inventory/efficientip.py

```diff
@@ -0,0 +1,360 @@
+        return False
+
+    def _populate(self, adevs):
+        for dev in adevs:
+            badded = False
+
+            # check if group
+            for _group in self.group_metadata:
+                if _group in dev and dev[_group] != '':
+                    _group_val = "{0}_{1}".format(_group, dev[_group])
+                    if _group_val not in self.group_metadata_val:
+                        self.inventory.add_group(_group_val)
+                        self.inventory.add_child(_group, _group_val)
+                    self.inventory.add_host(dev['name'], _group_val)
+
+                    badded = True
+
+            if not badded:
+                self.inventory.add_host(dev['name'])
+
+            if 'extracted_ips' in dev:
+                self.inventory.set_variable(dev['name'],
+                                            'ansible_host',
+                                            dev['extracted_ips'][0])
+
+            for filt in self.device_filter:
+                if filt['type'] == 'of_class':
+                    self.inventory.set_variable(dev['name'],
+                                                'eip_class',
+                                                filt['val'])
+                if filt['type'] == 'metadata':
+                    self.inventory.set_variable(dev['name'],
+                                                'eip_metadata_'
+                                                '{0}'.format(filt['name']),
+                                                filt['val'])
+
+    def parse(self, inventory, loader, path, cache=True):
+        super(InventoryModule, self).parse(inventory, loader, path, cache)
+        self.load_cache_plugin()
+
+        config = self._read_config_data(path)
+
+        cache_key = self.get_cache_key(path)
+        cache = self.get_option('cache')
+
+        cache_needs_update = True
+        adevs = []
+
+        if cache:
+            try:
+                adevs = self._cache[cache_key]
+                cache_needs_update = False
+            except KeyError:
+                cache_needs_update = True
+
+        if cache_needs_update:
+            # connects to the SOLIDserver
+            if self.sds_connect(config) is None:
+                return
+
+            # check the presence of the space
+            self.sds_switch_space(config)
+
+            # set filters
+            self.sds_set_filters(config)
+
+            # set group
+            self.sds_set_group(config)
+
+            adevs = sdsadv.list_devices(self.sds,
+                                        limit=self.limit,
+                                        filters=self.device_filter,
+                                        metadatas=self.group_metadata)
+
+        if cache_needs_update or (not cache and self.get_option('cache')):
+            self._cache[cache_key] = adevs
+
+        self._populate(adevs)
+
+#
+# ansible-test sanity --test pylint lib/ansible/plugins/inventory/efficientip.py
```

Comment:
Nice to add notes about testing needed.
Paths need updating

Reply:
updated